static/js/message_events.js

```diff
@@ -209,70 +260,21 @@ exports.update_messages = function update_messages(events) {
                         if (changed_narrow) {
                             const operators = new_filter.operators();
                             const opts = {
                                 trigger: 'stream/topic change',
                                 then_select_id: current_id,
                             };
                             narrow.activate(operators, opts);
-                            changed_narrow = true;
                         }
                     }
                 }
             }
-
-            const messages_to_rerender = [];
-            for (const id of event.message_ids) {
-                const msg = message_store.get(id);
-                if (msg === undefined) {
-                    continue;
-                }
-
-                // Remove the recent topics entry for the old topics;
-                // must be called before we call set_message_topic.
-                stream_topic_history.remove_message({
-                    stream_id: msg.stream_id,
-                    topic_name: msg.topic,
-                });
-
-                // Update the unread counts; again, this must be called
-                // before we modify the topic field on the message.
-                unread.update_unread_topics(msg, event);
-
-                // Now edit the attributes of our message object.
-                if (topic_edited) {
-                    msg.topic = new_topic;
-                    msg.topic_links = event.topic_links;
-                }
-                if (stream_changed) {
-                    const new_stream_name = stream_data.get_sub_by_id(new_stream_id).name;
-                    msg.stream_id = event.new_stream_id;
-                    msg.stream = new_stream_name;
-                    msg.display_recipient = new_stream_name;
-                }
-
-                // Add the recent topics entry for the new stream/topics.
-                stream_topic_history.add_message({
-                    stream_id: msg.stream_id,
-                    topic_name: msg.topic,
-                    message_id: msg.id,
-                });
-
-                if (!changed_narrow && current_filter && current_filter.can_apply_locally() &&
-                    !current_filter.predicate()(msg)) {
-                    // This topic edit makes this message leave the
-                    // current narrow, which is not being changed as
-                    // part of processing this event.  So we should
-                    // remove the message from the current/narrowed message list.
-                    const cur_row = current_msg_list.get_row(id);
-                    if (cur_row !== undefined) {
-                        messages_to_rerender.push({id: id});
-                    }
-                }
-            }
-            current_msg_list.remove_and_rerender(messages_to_rerender);
         }
 
         if (event.orig_content !== undefined) {
```

Comment:
This is mostly unrelated, but @cyphase I think given our discussion on edit history, I think we should replace most of the next block here, since we'll be going to the server for the edit history diff data in any case?